src/gui-qml/CMakeLists.txt

```
@@ -0,0 +1,32 @@
+# Target 'gui'
+add_library(
+  gui-qml
+  # Qt Resources
+  resources.qrc
```

```suggestion
  # Types for QML access
  ../gui/types.cpp
```

Two bits here.  First, if we link against the resources here, then we get a duplicate linkage error when we link against it again when we define the main executable in the primary CMake file.  Since we *must* link against it in the primary CMake file, we shouldn't link against it here.

The second issue that we run into is that the only file, besides the models, that we need from the main gui is the `types.cpp` file that enables accessing our custom types from QML.  I've done an ugly, direct linkage here, but I'm wondering if `types.cpp` should just be in the `models` folder and linked into that library.